### PR TITLE
Add new filter to parse xml output for network use cases

### LINKED ISSUE #31026: Add parse xml filter which returns a json object [XML_FILTER]
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

filter/network.py

##### ANSIBLE VERSION

```
2.5
```

##### SUMMARY

*  Add parse_xml filter to parse a xml response and convert to json object.

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

Fixes #31026
*  Add parse_xml filter
*  Add documentation for parse_xml filter

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

plugins/filter/network.py

##### ANSIBLE VERSION

```
2.5
```

##### ADDITIONAL INFORMATION

Playbook:
```
---
- name: parse xml output
  hosts: localhost
  vars:
    output: >
            <rpc-reply>
              <configuration>
                <vlans>
                  <vlan inactive="inactive">
                   <name>vlan-1</name>
                   <vlan-id>100</vlan-id>
                   <description>This is vlan-1</description>
                  </vlan>
                </vlans>
              </configuration>
            </rpc-reply>

  tasks:
  - name: filter xml output
    debug:
      msg: "{{ output | parse_xml('vlan_spec_xml.yml') }}"
```
Spec file:
```
#cat vlan_spec_xml.yml
---
vars:
  vlan:
    vlan_id: "{{ item.vlan_id }}"
    name: "{{ item.name }}"
    desc: "{{ item.desc }}"
    enabled: "{{ item.state.get('inactive') != 'inactive' }}"
    state: "{% if item.state.get('inactive') == 'inactive'%}inactive{% else %}active{% endif %}"

keys:
  vlans:
    type: list
    value: "{{ vlan }}"
    top: configuration/vlans/vlan
    items:
      vlan_id: vlan-id
      name: name
      desc: description
      state: ".[@inactive='inactive']"
```

Output:
```
TASK [filter xml output] ************************************************************************************
task path: /home/users/parse_xml_filter_vlan.yml:18
ok: [localhost] => {
    "msg": {
        "vlans": [
            {
                "desc": "This is vlan-1",
                "enabled": false,
                "name": "vlan-1",
                "state": "inactive",
                "vlan_id": 100
            }
        ]
    }
}
```

### CODE REVIEW

File: docs/docsite/rst/playbooks_filters.rst
Comment (privateip) [inline docs/docsite/rst/playbooks_filters.rst:485]: probably need to explain this a little more (usage of `.[@inactive='inactive']`)

**Review by privateip — CHANGES_REQUESTED** [review]
overall it looks good.  the only comment would be to add some notes around how to extract xml tags

File: docs/docsite/rst/playbooks_filters.rst
Comment (gundalow) [inline docs/docsite/rst/playbooks_filters.rst]: Should this read
> The spec file is a YAML file that defines

**Review by gundalow — CHANGES_REQUESTED** [review]
Would like to see some unit tests adding for this

**Comment by gundalow** [comment]
Once we are happy with the technical content it will need final review from @dharmabumstead

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Pleae revise this paragraph - it’s difficult to parse.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/playbooks_filters.rst
Comment (ganeshrn) [reply docs/docsite/rst/playbooks_filters.rst:485]: Fixed

**Comment by ganeshrn** [comment]
ready_for_review

**Comment by ganeshrn** [comment]
ready_for_review

File: docs/docsite/rst/playbooks_filters.rst
Comment (gundalow) [inline docs/docsite/rst/playbooks_filters.rst]: Typo relative

File: docs/docsite/rst/playbooks_filters.rst
Comment (gundalow) [inline docs/docsite/rst/playbooks_filters.rst]: Please replace wrt: with respect to

**Review by gundalow — CHANGES_REQUESTED** [review]

**Review by gundalow — APPROVED** [review]

**Review by privateip — APPROVED** [review]

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [reply docs/docsite/rst/playbooks_filters.rst]: Replace typo “relatvie” with “relative”
Fix “till inner-most container”

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [reply docs/docsite/rst/playbooks_filters.rst]: Capitalize “XML”

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Please revise to “The YAML spec file defines...”

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Hard to parse this sentence. Please rephrase. 

Please capitalize XML.

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Please capitalize XML. Please use definite articles (“the”).

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Please use “for exsmpleL instead of “i.e.”

File: docs/docsite/rst/playbooks_filters.rst
Comment (dharmabumstead) [inline docs/docsite/rst/playbooks_filters.rst]: Please capitalize “XML” throughout.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
Some edits required. Thanks @ganeshrn!

**Comment by ganeshrn** [comment]
ready_for_review

**Comment by ganeshrn** [comment]
ready_for_review

**Review by dharmabumstead — APPROVED** [review]
Docs OK with a few more copy edits. OK to merge if the networking team has tech reviewed and approved. Thanks @ganeshrn!

**Comment by ganeshrn** [comment]
@privateip @gundalow @dharmabumstead Thank you for reviewing.

**Pull request merged**
